backends/mock/register.go

```diff
@@ -0,0 +1,15 @@
+package mock
+
+import (
+	"github.com/bndw/pick/backends"
+)
+
+const (
+	// Make sure to not export the mock client name
+	clientName     = "_mock"
```

Comment:
Exposing the mock client name allows you to construct a mock backend via `backends.NewWithType(CLIENT_NAME, config)`, but one should use `mock.NewForTesting` instead to avoid having to check for an error returned by `backends.NewWithType`.
Not exporting `clientName` prevents us from accidentally doing that. However, you still get an instance by hard-coding the mock client name: `backends.NewWithType("_mock", config)`. This could be avoided by letting `backends.NewWithType` accept an empty interface as its client name, see this example:

```go
type internal string

const clientName internal = "_mock"

backends.Register(clientName, ..) // Now accepts an interface{} as first argument

// Now, no other package can request a mock client instance, as the "internal" type is not exported
```

Would you prefer something like this?

Reply:
I'm good with the current implementation, thanks for explaining.